src/components/macroview/rightPanel/editForms/DelayForm.tsx

```diff
@@ -13,9 +28,15 @@ export default function DelayForm({
   selectedElementId,
   selectedElement
 }: Props) {
-  const [delayDuration, setDelayDuration] = useState("0")
+  const config = useSettingsContext()
+  const [delayDuration, setDelayDuration] = useState(
```

Comment:
again, using default duration from the config

Reply:
Is it possible for the config to get changed when inside this form? I guess it isn't, but if it is, then you may need to handle the change with a useEffect

Reply:
I think it shouldn't but at the same time there might be a case in 1.2 where something of that sort could happen.

Something like this should solve it?:

```ts
const [config, setConfig] =  useState<SettingsState>(useSettingsContext());
  const newSettings = useSettingsContext()

  useEffect(() => {
    setConfig(newSettings)
  }, [newSettings])
```

Reply:
For this situation, it wouldn't make sense to do it on the basis of the settings. You would do this approach, but just for the delayDuration

Reply:
Thought about it, but I think in newer versions when we have migration of settings, I'll have a default duration either separate for each macro, or just keep the default and it won't change mid-macro edit. Would be very annoying if the users just had to keep changing it